src/components/pokemon/pokemon.mapper.ts

```diff
@@ -1,11 +1,13 @@
-import { Pokemon } from './pokemon.model';
+import { CPCalculator } from '@core/cp-calculator';
+import { DexParser } from './shared/dex-parser.component';
+import { GenderPercentParser } from './shared/gender-percent-parser.component';
 import { ItemTemplate } from '@core/game_master/gameMaster';
+import { Pokemon } from './pokemon.model';
+import { RootObject } from '@core';
 import { Util } from '@util';
-import { CPCalculator } from '../../core/cp-calculator';
-import { DexParser } from './shared/dex-parser.component';
 
 export class PokemonMapper {
-    public static Map(rawPokemon: ItemTemplate): Pokemon {
+    public static Map(rawPokemon: ItemTemplate, gameMaster: RootObject): Pokemon {
```

Comment:
It is consistent with JavaScript's view of `map`, but not with the generic view of `map`. I personally don't really see anything controversial about it. I would say that making the evolution parsing code depend on the RootObject rather than the parsed list of pokemon will take some extra code.